File: src/main/java/org/stevenlowes/tools/lifx_controller/Messages/Device/EchoRequest.java
package org.stevenlowes.tools.lifx_controller.Messages.Device;

import java.util.Random;

import org.stevenlowes.tools.lifx_controller.Messages.DataTypes.Payload;

public class EchoRequest extends Payload{
	private final int code = 58;
	private byte[] payload;			// 64-Bytes
	
	public EchoRequest() {
		payload = new byte[64];
		new Random().nextBytes(payload);
	}
	
	public EchoRequest(byte[] payload) {
		this.payload = payload;
	}
	
	public EchoRequest(EchoRequest echoRequest) {
		payload = echoRequest.payload;
	}
	
	public int getCode() {
		return code;
	}
	
	public byte[] getPayload() {
		return payload;
	}
	
	public void setPayload(byte[] payload) {
		this.payload = payload;
	}
	
	public byte[] getByteArray() {
		byte[] byteArray = new byte[64];
		for(int i=0; i<64; i++) byteArray[i] = payload[(-1*i)+63];
		return byteArray;
	}

}

File: src/main/java/org/stevenlowes/tools/lifxcontroller/commands/request/device/EchoRequest.kt
package org.stevenlowes.tools.lifxcontroller.commands.request.device

import org.stevenlowes.tools.lifxcontroller.commands.request.RequestCommand
import java.util.*

data class EchoRequest(val payload: ByteArray = ByteArray(64)) : RequestCommand(58) {
    override val payloadBytes: ByteArray = ByteArray(64)

    init {
        for (i in 0..63) {
            payloadBytes[i] = payload[-1 * i + 63]
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as EchoRequest

        if (!Arrays.equals(payload, other.payload)) return false
        if (!Arrays.equals(payloadBytes, other.payloadBytes)) return false

        return true
    }

    override fun hashCode(): Int {
        var result = Arrays.hashCode(payload)
        result = 31 * result + Arrays.hashCode(payloadBytes)
        return result
    }
}
